apps/prairielearn/elements/pl-number-input/pl-number-input.py

```diff
@@ -210,10 +213,17 @@ def render(element_html, data):
             info_params.pop("format", None)
             # Within mustache, the shortformat generates the shortinfo that is used as a placeholder inside of the numeric entry.
             # Here we opt to not generate the value, hence the placeholder is empty.
-            info_params["shortformat"] = pl.get_boolean_attrib(
-                element, "show-placeholder", SHOW_PLACEHOLDER_DEFAULT
-            )
-            shortinfo = chevron.render(f, info_params).strip()
+            # The placeholder text may be overriden by setting the 'placeholder' attribute in the pl-number-input HTML tag
```

Comment:
I think this can be done more cleanly, see https://github.com/PrairieLearn/PrairieLearn/pull/6980/files#diff-30a7667069c2672e849167650819b4eb9fc32c03fea4775a2810f1530a5b96a9R107-R112

Reply:
Yeah, the refactored approach from `pl-integer-input` is definitely way nicer. I tried my hand at doing something similar in the latest commit. Could you have a look when you have the time.

Reply:
I agree it looks much cleaner now! I think we have to keep the `show-placeholder` attribute for backward compatibility, so I think this is about as clean as we can get it. Should be good to start adding more changes to this PR.